docs/h5py_dataset.rst

```diff
@@ -105,11 +111,73 @@ with training examples and the last 10 rows with test examples.
+>>> split_array[2:6:3]['source'] = 'targets'.encode('utf8')
+>>> split_array[0:3]['start'] = 0
+>>> split_array[0:3]['stop'] = 90
+>>> split_array[3:6]['start'] = 90
+>>> split_array[3:6]['stop'] = 100
+>>> split_array[:]['available'] = True
+>>> split_array[:]['comment'] = '.'.encode('utf8')
+>>> f.attrs['split'] = split_array
+
+We created a 1D numpy array with six elements. The ``dtype`` for this array
+is a compound type: every element of the array is a tuple of ``(str, str, int,
+int, bool, str)``. The length of each string element has been chosen to be the
+maximum length we needed to store: that's 5 for the ``split`` element
+(``'train'`` being the longest split name) and 15 for the ``source`` element
+(``'vector_features'`` being the longest source name). We didn't include any
+comment, so the length for that element was set to 1. Due to a quirk in pickling
+empty strings, we put ``'.'`` as the comment value.
 
->>> f.attrs['train'] = [0, 90]
->>> f.attrs['test'] = [90, 100]
+.. warning::
+
+    Due to limitations in h5py, you must make sure to use bytes for ``split``,
+    ``source`` and ``comment``.
+
+:class:`~.datasets.hdf5.H5PYDataset` expects the ``split`` attribute of the
+root node to contain as many elements as the cartesian product of all sources
+and all splits, *i.e.* all possible split/source combinations. Sometimes, no
+data is available for some source/split combination: for instance, the test
+set may not be labeled, and the ``('test', 'targets')`` combination may not
+exist. In that case, you can set the ``available`` element for that combination
+to ``False``, and :class:`~.datasets.hdf5.H5PYDataset` will ignore it.
+
+The method described above does the job, but it's not very convenient. An even
+simpler way of achieving the same result is to call
+:meth:`~.datasets.hdf5.H5PYDataset.create_split_array`.
+
+>>> from fuel.datasets.hdf5 import H5PYDataset
+>>> split_dict = {
+...     'train': {'vector_features': (0, 90), 'image_features': (0, 90),
+...               'targets': (0, 90)},
+...     'test': {'vector_features': (90, 100), 'image_features': (90, 100),
+...              'targets': (90, 100)}}
+>>> f.attrs['split'] = H5PYDataset.create_split_array(split_dict)
+
+The :meth:`~.datasets.hdf5.H5PYDataset.create_split_array` method expects
+a dictionary mapping split names to dictionaries. Those dictionaries map source
+names to tuples of length 2 or 3: the first two elements correspond to the start
+and stop indexes and the last element is an optional comment for the
+split/source pair. The method will create the array behind the scenes,
+choose the string lengths automatically and populate it with the information
+in the split dictionary. If a particular split/source combination isn't present,
+its ``available`` attribute is set to ``False``, which allows us to specify
+only what's actually present in the HDF5 file we created.
 
 We flush, close the file and *voilà*!
```

Comment:
I guess it should be, it's fine with unicode characters in docstrings as long as you use `u""""`.